### PR TITLE
chore: Introduce release notes formatting + preview release action

### PR DESCRIPTION
**Pull request by olivermrbl**
### What
- Add changelog formatting file to automate pull request categorization in release notes*
- Add action to trigger preview releases
- Remove unused staging deployment action

**Add changelog formatting file to automate pull request categorization in release notes**

In order for this work, we need to be more strict about our labels usage. More specifically, we need to add one of the four different labels noted in the `release.yml` file. The labels should be self-explanatory. If your PR introduces both features and bugs, you should choose the label based on importance of the changes.

### CODE REVIEW

File: .github/release.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        - "type: bug"
+    - title: Documentation
+      labels:
+        - "type: docs"
+    - title: Chores
+      labels:
+        - "type: chore"
+    - title: Other Changes
+      labels:
+        - "*"
```
Comment (adrien2p) [inline .github/release.yml:20]: Where those labels must be specified in the pr? is it in the change set?

Comment (olivermrbl) [reply .github/release.yml:20]: Labels on the PR

Comment (adrien2p) [inline .github/release.yml:20]: I think we could add enhancement as well or improvements and performance wdyt? Those are things that we often encounter

**Review by adrien2p — APPROVED** [review]
LGTM, one small suggestion

File: .github/release.yml (hunk 1 above)
Comment (olivermrbl) [reply .github/release.yml:20]: Discussed internally. We will not be adding this right now.

**Pull request merged**